x-pack/plugins/screenshotting/server/formats/pdf/pdf_maker/index.ts

```diff
@@ -26,37 +26,34 @@ export async function pngsToPdf({
   logo,
   title,
   packageInfo,
-  logger,
+  eventLogger,
 }: PngsToPdfArgs): Promise<{ buffer: Buffer; pages: number }> {
-  const pdfMaker = new PdfMaker(layout, logo, packageInfo, logger);
-  const tracker = getTracker();
+  eventLogger.pdfStart();
+  const pdfMaker = new PdfMaker(layout, logo, packageInfo, eventLogger.kbnLogger);
   if (title) {
     pdfMaker.setTitle(title);
   }
   results.forEach((result) => {
     result.screenshots.forEach((png) => {
-      tracker.startAddImage();
+      eventLogger.addPdfImageStart();
       pdfMaker.addImage(png.data, {
         title: png.title ?? undefined,
         description: png.description ?? undefined,
       });
-      tracker.endAddImage();
+      eventLogger.addPdfImageEnd();
     });
   });
 
   let buffer: Uint8Array | null = null;
   try {
-    tracker.startCompile();
+    eventLogger.compilePdfStart();
     buffer = await pdfMaker.generate();
-    tracker.endCompile();
+    eventLogger.compilePdfEnd();
```

Comment:
Shall we handle the case when "compile PDF" fails?

Reply:
added in 1ed749bfedf